blobstore/src/main/java/org/jclouds/blobstore/TransientStorageStrategy.java

```diff
@@ -148,8 +149,13 @@ public boolean blobExists(final String containerName, final String blobName) {
    }
 
    @Override
-   public Iterable<String> getBlobKeysInsideContainer(final String containerName) {
-      return containerToBlobs.get(containerName).keySet();
+   public Iterable<String> getBlobKeysInsideContainer(final String containerName, String prefix) {
+      ConcurrentSkipListMap<String, Blob> blobs = containerToBlobs.get(containerName);
+      if (prefix == null) {
+         return blobs.keySet();
+      }
+      String lastPrefix = prefix + (char) 65535;  // TODO: better sentinel?
```

Comment:
This is hacky but I don't think is has practical drawbacks?

Reply:
[minor] Perhaps create a constant `SENTINEL` or so just to make it clear that that's what the last character is supposed to be?

That might be confusing, especially if the TODO is removed in future?